src/test.c

```diff
@@ -28,6 +30,45 @@ struct Table* retrieve(struct ParseTree* tree) {
   return NULL;
 }
 
+void testCreateTable(void) {
+
+  struct ParseTree* createDBCommand = createCreateDatabaseParseTree("foo");
+  createDatabase(createDBCommand);
+  assert(strcmp(currentDatabase, "foo") == 0, "currentDatabase should be set!");
+  char tableFolderPath[1000];
+
+  char names[FIELD_SIZE][NAME_LIMIT] = {
+    "name1",
+    "name2",
+    "name3",
+    "name4",
+  };
+
+  char values[FIELD_SIZE][VALUE_LIMIT] = {
+    "value1",
+    "value2",
+    "value3",
+    "value4",
+  };
+  struct Field* fields = createFieldList(names, values, 4);
+
+  struct ParseTree* createTableCmd = createCreateTableParseTree("bar", fields);
+  createTable(createTableCmd);
+  sprintf(tableFolderPath, "%s/foo/bar", DATABASE_DIR);
+  assert(access(tableFolderPath, F_OK) != -1, "Table file was not constructed!");
+
+  char fileContents[1000];
+  FILE* file = fopen(tableFolderPath, "r");
+  fgets(fileContents, 999, file);
+
+  assert(strcmp(fileContents, "name1|name2|name3|name4\n") == 0, "Table was not written correctly!");
```

Comment:
Just to point it out, this is essentially the format of the table data file. I ended up choosing PIPE(`|`) because I think it looks the cleanest, and I had to pick something.

Also, right off the bat I noticed that type information isn't persisted anywhere here, which means I'll need to 1. have the field struct carry around types as well, and 2. somehow encode that into here. (Maybe something like fieldname[TYPE]) That way when we do `SELECT`s later, we can know the types!